source/js/components/search-form.js

```diff
@@ -0,0 +1,108 @@
+import fetchUtils from "@degjs/fetch-utils";
+
+const searchForm = () => {
+
+
+    const formEl = document.querySelector('.js-search-form');
+    const layoutEl = document.querySelector('.js-general-content');
+    const searchFieldSet = document.querySelector('.js-search-fieldset');
+    let searchInputEl;
+    const searchButtonEl = formEl.querySelector('.js-search-btn');
+
+    const url = formEl.dataset.api;
+
+    function init() {
+        searchInputEl = formEl.querySelector('.js-search');
+        formEl.addEventListener('submit', searchSubmit);
+    }
+
+    function searchSubmit(e) {
+        e.preventDefault();
+        getData();
+    }
+
+    function getData() {
+        preloader(true);
+        fetchUtils.getJSON(
+            url,
+            {
+                body: JSON.stringify(searchInputEl.value),
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            }
+        )
+        .then((data) => {
```

Comment:
[suggestion] The parens around `data` are unnecessary, since you only have one argument